src/App.tsx

```diff
@@ -15,36 +16,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'Sort alphabetically',
+  ByLength = 'Sort by length',
+  Default = 'Default',
+}
+
+type SortField = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+const SORT_FIELD_ALPHABETICALLY: SortField = {
+  sortType: SortType.Alphabetically,
+  isReversed: false,
+};
+
+const SORT_FIELD_BY_LENGTH: SortField = {
+  sortType: SortType.ByLength,
+  isReversed: false,
+};
+
+function getPreparedGoods(goods: string[], sortField: SortField) {
+  const preparedGoods = [...goods];
+
+  switch (sortField.sortType) {
+    case SORT_FIELD_ALPHABETICALLY.sortType:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SORT_FIELD_BY_LENGTH.sortType:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (sortField.isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField>({
+    sortType: SortType.Default,
+    isReversed: false,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() =>
+            setSortField(prevState => ({
+              sortType: SortType.Alphabetically,
+              isReversed: prevState.isReversed,
+            }))
```

Comment:
```suggestion
            setSortField(prevState => ({
              ...prevState,
              sortType: SortType.Alphabetically,
            }))
```
You can copy all previous values and just update what you need